core/src/main/scala/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -229,10 +255,60 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
     }
     val resolver = new MetaResolver(source.table)
     val tiColumns: Seq[TiColumnRef] = extractTiColumnRefFromExpressions(tiColumnSet)
-    tiColumns.foreach { resolver.resolve(_) }
+    tiColumns.foreach {
+      resolver.resolve(_)
+    }
     tiColumns
   }
 
+  // If we have a partition info such as
+  // partition by year(`purchased`)
+  // p0 less than 1990
+  // p1 less than 2000,
+  // we will get
+  // p0 year(`purchased`) < 1990
+  // p1 year(`purchased`) >= 1990 and year(`purchased`) < 2000
+  private def generatePartitionExpr(pInfo: TiPartitionInfo): TiPartitionExpr = {
+    val parser: SparkSqlParser = new SparkSqlParser(sparkSession.sqlContext.conf)
```

Comment:
Why does Spark Parser works here? Don't you need your own tiny parser?

Reply:
SparkSqlParser is enough here. 
column used in partition expression will never be an alias so we can parse it first and then resolve it later by taking advantage of MetaResolver.resolve signature.